xtask/src/main.rs

```diff
@@ -83,11 +79,28 @@ enum Task {
 fn execute_task() -> Result<()> {
     let cli = Cli::parse();
     match cli.task {
-        Task::InstallTools => install_tools()?,
-        Task::WebTests { dir } => run_web_tests(dir)?,
-        Task::RustTests => run_rust_tests()?,
-        Task::Serve { language, output } => start_web_server(language, output)?,
-        Task::Build { language, output } => build(language, output)?,
+        Task::InstallTools => install_tools(),
+        Task::WebTests { dir } => run_web_tests(dir),
+        Task::RustTests => run_rust_tests(),
+        Task::Serve { language, output } => start_web_server(language, output),
+        Task::Build { language, output } => build(language, output),
+    }
+}
+
+/// Executes a command and returns an error if it fails.
+fn run_command(cmd: &mut Command) -> Result<()> {
```

Comment:
I have a hunch that `run_command` should take `Command` instead of a reference: I don't think you have a case where you want to run the command twice. But you can also keep it like this if you find it more flexible.